pkg/server/server.go

```diff
@@ -96,15 +96,20 @@ func (k *KiamServer) GetPodCredentials(ctx context.Context, req *pb.GetPodCreden
 
 	if !decision.IsAllowed() {
 		logger.WithField("policy.explanation", decision.Explanation()).Errorf("pod denied by policy")
+		k.recordEvent(pod, v1.EventTypeWarning, "KiamRoleForbidden",
```

Comment:
In this event we should also be able to use `decision.Explanation()` to include more information about why it was rejected.